importer gets stuck with DEMO bucket streams

<!--  Thanks for submitting a bug report!  Before submitting:
1. Try searching the existing issues to see if your issue has already been reported
2. Try asking on our Discord group (https://discordapp.com/invite/FFb9YFX)
3. If you're reporting a security vulnerability, please email security@hedera.com instead of opening an issue
-->

**Detailed Description**

With the latest 0.16.0-SNAPSHOT docker environment and the DEMO bucket streams, the importer will get stuck when >= 2 groups of signature files fall into the same bucket.

**Actual Behavior**
Steps to reproduce the behavior:
1. check out latest master
2. ./mvnw install -DskipTests
3. docker-compose up -d
4. docker logs -f hedera-mirror-node_importer_1

The log messages contain a log of HashMismatchException errors:

```
2020-07-27 16:21:12,800 ERROR [scheduling-3] c.h.m.i.d.r.RecordFileDownloader Error downloading data file from node 0.0.4 corresponding to 2019-10-11T16_17_50.389374Z.rcd_sig. Will retry another node
java.lang.IllegalArgumentException: Error parsing bad record file 2019-10-11T16_17_50.389374Z.rcd
	at com.hedera.mirror.importer.util.Utility.parseRecordFile(Utility.java:217) ~[classes/:?]
	at com.hedera.mirror.importer.downloader.record.RecordFileDownloader.verifyDataFile(RecordFileDownloader.java:79) ~[classes/:?]
	at com.hedera.mirror.importer.downloader.Downloader.verifySigsAndDownloadDataFiles(Downloader.java:403) [classes/:?]
	at com.hedera.mirror.importer.downloader.Downloader.downloadNextBatch(Downloader.java:144) [classes/:?]
	at com.hedera.mirror.importer.downloader.record.RecordFileDownloader.download(RecordFileDownloader.java:58) [classes/:?]
	at jdk.internal.reflect.GeneratedMethodAccessor123.invoke(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.springframework.scheduling.support.ScheduledMethodRunnable.run(ScheduledMethodRunnable.java:84) [spring-context-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) [spring-context-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: com.hedera.mirror.importer.exception.HashMismatchException: Hash mismatch for file {}. Expected = {}, Actual = {}
	at com.hedera.mirror.importer.util.Utility.parseRecordFile(Utility.java:173) ~[classes/:?]
	... 15 more
```

**Expected Behavior**
Importer was able to import and parse all files from the DEMO bucket and there shouldn't be any HashMismatchExceptions.

**Environment:**
- Java: [e.g. OpenJDK 11.0.4]
- Node: [e.g. v12.9.1-x86]
- OS: [e.g. Ubuntu 18.04]
- Version: [e.g. 1.0.1]

**Additional Context**

--- a/docs/configuration.md
+++ b/docs/configuration.md
@@ -29,7 +29,6 @@ value, it is recommended to only populate overridden properties in the custom `a
 | `hedera.mirror.importer.db.username`                                 | mirror_node             | The username the processor uses to connect to the database                                     |
 | `hedera.mirror.importer.downloader.accessKey`                        | ""                      | The cloud storage access key                                                                   |
 | `hedera.mirror.importer.downloader.balance.batchSize`                | 15                      | The number of signature files to download per node before downloading the signed files         |
-| `hedera.mirror.importer.downloader.balance.closeInterval`            | 15m                     | The expected interval between balance files                                                    |
 | `hedera.mirror.importer.downloader.balance.enabled`                  | true                    | Whether to enable balance file downloads                                                       |
 | `hedera.mirror.importer.downloader.balance.frequency`                | 30s                     | The fixed period between invocations. Can accept duration units like `10s`, `2m` etc.          |
 | `hedera.mirror.importer.downloader.balance.keepSignatures`           | false                   | Whether to keep balance signature files after successful verification. If false, files are deleted. |
@@ -39,7 +38,6 @@ value, it is recommended to only populate overridden properties in the custom `a
 | `hedera.mirror.importer.downloader.cloudProvider`                    | S3                      | The cloud provider to download files from. Either `S3` or `GCP`                                |
 | `hedera.mirror.importer.downloader.endpointOverride`                 |                         | Can be specified to download streams from a source other than S3 and GCP. Should be S3 compatible |
 | `hedera.mirror.importer.downloader.event.batchSize`                  | 100                     | The number of signature files to download per node before downloading the signed files         |
-| `hedera.mirror.importer.downloader.event.closeInterval`              | 5s                      | The expected interval between event files                                                      |
 | `hedera.mirror.importer.downloader.event.enabled`                    | false                   | Whether to enable event file downloads                                                         |
 | `hedera.mirror.importer.downloader.event.frequency`                  | 5s                      | The fixed period between invocations. Can accept duration units like `10s`, `2m` etc.          |
 | `hedera.mirror.importer.downloader.event.keepSignatures`             | false                   | Whether to keep event signature files after successful verification. If false, files are deleted. |
@@ -48,7 +46,6 @@ value, it is recommended to only populate overridden properties in the custom `a
 | `hedera.mirror.importer.downloader.gcpProjectId`                     |                         | GCP project id to bill for requests to GCS bucket which has Requester Pays enabled.            |
 | `hedera.mirror.importer.downloader.maxConcurrency`                   | 1000                    | The maximum number of allowed open HTTP connections. Used by AWS SDK directly.                 |
 | `hedera.mirror.importer.downloader.record.batchSize`                 | 40                      | The number of signature files to download per node before downloading the signed files         |
-| `hedera.mirror.importer.downloader.record.closeInterval`             | 2s                      | The expected interval between record files                                                     |
 | `hedera.mirror.importer.downloader.record.enabled`                   | true                    | Whether to enable record file downloads                                                        |
 | `hedera.mirror.importer.downloader.record.frequency`                 | 500ms                   | The fixed period between invocations. Can accept duration units like `10s`, `2m` etc.          |
 | `hedera.mirror.importer.downloader.record.keepSignatures`            | false                   | Whether to keep record signature files after successful verification. If false, files are deleted. |

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/Downloader.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/Downloader.java
@@ -151,9 +151,9 @@ public abstract class Downloader {
 
     /**
      * Download all signature files with a timestamp later than the last valid file. Put signature files into a
-     * multi-map sorted and grouped by the expected closing interval.
+     * multi-map sorted and grouped by the timestamp.
      *
-     * @return a multi-map of expected close interval to signature file objects from different nodes
+     * @return a multi-map of signature file objects from different nodes, grouped by timestamp
      */
     private Multimap<Long, FileStreamSignature> downloadSigFiles() throws InterruptedException {
         String lastValidFileName = applicationStatusRepository.findByStatusCode(getLastValidDownloadedFileKey());
@@ -168,8 +168,6 @@ public abstract class Downloader {
                 .collect(Collectors.toSet());
         List<Callable<Object>> tasks = new ArrayList<>(nodeAccountIds.size());
         var totalDownloads = new AtomicInteger();
-        long closeInterval = downloaderProperties.getCloseInterval().toNanos();
-        long lastValidTimestamp = Utility.getTimestampFromFilename(lastValidFileName);
         log.info("Downloading signature files created after file: {}", lastValidSigFileName);
 
         /**
@@ -223,15 +221,8 @@ public abstract class Downloader {
                                 FileStreamSignature fileStreamSignature = new FileStreamSignature();
                                 fileStreamSignature.setFile(sigFile);
                                 fileStreamSignature.setNode(nodeAccountId);
-                                long groupTimestamp = getGroupTimestamp(lastValidTimestamp, closeInterval,
-                                        fileStreamSignature);
-
-                                if (groupTimestamp > 0) {
-                                    sigFilesMap.put(groupTimestamp, fileStreamSignature);
-                                } else {
-                                    log.warn("Ignoring signature associated with the previously processed stream " +
-                                            "file: {}", fileStreamSignature);
-                                }
+                                long timestamp = Utility.getTimestampFromFilename(sigFile.getName());
+                                sigFilesMap.put(timestamp, fileStreamSignature);
                             }
                         } catch (InterruptedException ex) {
                             log.warn("Failed downloading {} in {}", pendingDownload.getS3key(),
@@ -256,35 +247,6 @@ public abstract class Downloader {
             log.info("Downloaded {} signatures in {} ({}/s)", totalDownloads, stopwatch, rate);
         }
         return sigFilesMap;
-    }
-
-    /**
-     * Partitions signature files into a time range [T + I/2, T + I * 3/2) where T is the last valid timestamp and I is
-     * the stream close interval. Calculates the start time of the nearest range that includes the timestamp of the
-     * current file.
-     *
-     * @param lastValidTimestamp the last timestamp of the signature that was successfully verified
-     * @param closeInterval      how often the file closes
-     * @param signature          domain object
-     * @return a timestamp representing the start time of the bucket this signature should be associated with
-     */
-    private long getGroupTimestamp(long lastValidTimestamp, long closeInterval, FileStreamSignature signature) {
-        long currentTimestamp = Utility.getTimestampFromFilename(signature.getFile().getName());
-        long groupTimestamp = 0;
-
-        // Initial file has no previous so we can't calculate offset from that
-        if (lastValidTimestamp <= 0) {
-            groupTimestamp = currentTimestamp;
-        } else {
-            double interval = (double) (currentTimestamp - lastValidTimestamp);
-            long bucket = Math.round(interval / closeInterval);
-
-            if (bucket > 0) {
-                groupTimestamp = lastValidTimestamp + bucket * closeInterval + closeInterval / 2;
-            }
-        }
-
-        return groupTimestamp;
     }
 
     /**
@@ -362,17 +324,24 @@ public abstract class Downloader {
         NodeSignatureVerifier nodeSignatureVerifier = new NodeSignatureVerifier(networkAddressBook);
         Path validPath = downloaderProperties.getValidPath();
 
-        for (Long groupId : sigFilesMap.keySet()) {
+        for (var groupIdIterator = sigFilesMap.keySet().iterator(); groupIdIterator.hasNext(); ) {
             if (ShutdownHelper.isStopping()) {
                 return;
             }
 
             Instant startTime = Instant.now();
+            long groupId = groupIdIterator.next();
             Collection<FileStreamSignature> signatures = sigFilesMap.get(groupId);
             boolean valid = false;
 
             try {
                 nodeSignatureVerifier.verify(signatures);
+            } catch (SignatureVerificationException ex) {
+                if (groupIdIterator.hasNext()) {
+                    log.warn("Signature verification failed but still have files in the batch, try to process the next group", ex);
+                    continue;
+                }
+                throw ex;
             } finally {
                 for (FileStreamSignature signature : signatures) {
                     EntityId nodeAccountId = EntityId.of(signature.getNode(), EntityTypeEnum.ACCOUNT);
@@ -425,7 +394,7 @@ public abstract class Downloader {
             }
 
             if (!valid) {
-                log.error("File could not be verified by at least 1/3 of nodes: {}", signatures);
+                log.error("None of the data files could be verified, signatures: {}", signatures);
             }
 
             streamVerificationMetric.tag("success", String.valueOf(valid))

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/Downloader.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/Downloader.java
@@ -153,14 +153,14 @@ public abstract class Downloader {
      * Download all signature files with a timestamp later than the last valid file. Put signature files into a
      * multi-map sorted and grouped by the timestamp.
      *
-     * @return a multi-map of signature file objects from different nodes, grouped by timestamp
+     * @return a multi-map of signature file objects from different nodes, grouped by filename
      */
-    private Multimap<Long, FileStreamSignature> downloadSigFiles() throws InterruptedException {
+    private Multimap<String, FileStreamSignature> downloadSigFiles() throws InterruptedException {
         String lastValidFileName = applicationStatusRepository.findByStatusCode(getLastValidDownloadedFileKey());
         // foo.rcd < foo.rcd_sig. If we read foo.rcd from application stats, we have to start listing from
         // next to 'foo.rcd_sig'.
         String lastValidSigFileName = lastValidFileName.isEmpty() ? "" : lastValidFileName + "_sig";
-        Multimap<Long, FileStreamSignature> sigFilesMap = Multimaps
+        Multimap<String, FileStreamSignature> sigFilesMap = Multimaps
                 .synchronizedSortedSetMultimap(TreeMultimap.create());
         Set<String> nodeAccountIds = networkAddressBook.getAddresses()
                 .stream()
@@ -221,8 +221,7 @@ public abstract class Downloader {
                                 FileStreamSignature fileStreamSignature = new FileStreamSignature();
                                 fileStreamSignature.setFile(sigFile);
                                 fileStreamSignature.setNode(nodeAccountId);
-                                long timestamp = Utility.getTimestampFromFilename(sigFile.getName());
-                                sigFilesMap.put(timestamp, fileStreamSignature);
+                                sigFilesMap.put(sigFile.getName(), fileStreamSignature);
                             }
                         } catch (InterruptedException ex) {
                             log.warn("Failed downloading {} in {}", pendingDownload.getS3key(),
@@ -320,7 +319,7 @@ public abstract class Downloader {
      * the data file into `valid` directory; else download the data file from other valid node folder and compare the
      * hash until we find a match.
      */
-    private void verifySigsAndDownloadDataFiles(Multimap<Long, FileStreamSignature> sigFilesMap) {
+    private void verifySigsAndDownloadDataFiles(Multimap<String, FileStreamSignature> sigFilesMap) {
         NodeSignatureVerifier nodeSignatureVerifier = new NodeSignatureVerifier(networkAddressBook);
         Path validPath = downloaderProperties.getValidPath();
 
@@ -330,7 +329,7 @@ public abstract class Downloader {
             }
 
             Instant startTime = Instant.now();
-            long groupId = groupIdIterator.next();
+            String groupId = groupIdIterator.next();
             Collection<FileStreamSignature> signatures = sigFilesMap.get(groupId);
             boolean valid = false;
 

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/DownloaderProperties.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/DownloaderProperties.java
@@ -32,10 +32,6 @@ public interface DownloaderProperties {
 
     int getBatchSize();
 
-    Duration getCloseInterval();
-
-    void setCloseInterval(Duration closeInterval);
-
     CommonDownloaderProperties getCommon();
 
     MirrorProperties getMirrorProperties();

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/NodeSignatureVerifier.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/NodeSignatureVerifier.java
@@ -52,7 +52,7 @@ public class NodeSignatureVerifier {
                 .collect(Collectors.toMap(NodeAddress::getId, NodeAddress::getPublicKeyAsObject));
     }
 
-    private static boolean consensusReached(long actualNodes, long expectedNodes) {
+    private static boolean canReachConsensus(long actualNodes, long expectedNodes) {
         return actualNodes >= Math.ceil(expectedNodes / 3.0);
     }
 
@@ -72,7 +72,7 @@ public class NodeSignatureVerifier {
 
         final long sigFileCount = signatures.size();
         final long nodeCount = nodeIDPubKeyMap.size();
-        if (!consensusReached(sigFileCount, nodeCount)) {
+        if (!canReachConsensus(sigFileCount, nodeCount)) {
             throw new SignatureVerificationException("Require at least 1/3 signature files to reach consensus, got " +
                     sigFileCount + " out of " + nodeCount + " for file " + filename + ": " + statusMap(signatures));
         }
@@ -96,7 +96,7 @@ public class NodeSignatureVerifier {
         for (String key : signatureHashMap.keySet()) {
             Collection<FileStreamSignature> validatedSignatures = signatureHashMap.get(key);
 
-            if (consensusReached(validatedSignatures.size(), nodeIDPubKeyMap.size())) {
+            if (canReachConsensus(validatedSignatures.size(), nodeIDPubKeyMap.size())) {
                 consensusCount += validatedSignatures.size();
                 validatedSignatures.forEach(s -> s.setStatus(SignatureStatus.CONSENSUS_REACHED));
             }

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/NodeSignatureVerifier.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/NodeSignatureVerifier.java
@@ -23,6 +23,7 @@ package com.hedera.mirror.importer.downloader;
 import com.google.common.collect.HashMultimap;
 import com.google.common.collect.Multimap;
 import com.google.common.collect.Sets;
+import java.io.File;
 import java.security.PublicKey;
 import java.security.Signature;
 import java.util.Collection;
@@ -57,17 +58,21 @@ public class NodeSignatureVerifier {
     }
 
     /**
-     * Verifies that the signature files are signed by corresponding node's PublicKey. For valid signature files, we
-     * compare their hashes to see if at least 1/3 with the same filename have hashes that match. If a signature is
-     * valid, we put the hash in its content and its file to the map, to see if at least 1/3 valid signatures have the
-     * same hash.
+     * Verifies that the signature files satisfy the consensus requirement:
+     * <ol>
+     *  <li>At least 1/3 signature files are present</li>
+     *  <li>For a signature file, we validate it by checking if it's signed by corresponding node's PublicKey. For valid
+     *      signature files, we compare their hashes to see if at least 1/3 have hashes that match. If a signature is
+     *      valid, we put the hash in its content and its file to the map, to see if at lest 1/3 valid signatures have
+     *      the same hash</li>
+     * </ol>
      *
-     * @param signatures a list of a sig files which have the same timestamp
+     * @param signatures a list of signature files which have the same filename
      * @throws SignatureVerificationException
      */
     public void verify(Collection<FileStreamSignature> signatures) throws SignatureVerificationException {
         Multimap<String, FileStreamSignature> signatureHashMap = HashMultimap.create();
-        String filename = !signatures.isEmpty() ? signatures.stream().findFirst().get().getFile().getName() : null;
+        String filename = signatures.stream().map(FileStreamSignature::getFile).map(File::getName).findFirst().orElse(null);
         int consensusCount = 0;
 
         final long sigFileCount = signatures.size();
@@ -96,7 +101,7 @@ public class NodeSignatureVerifier {
         for (String key : signatureHashMap.keySet()) {
             Collection<FileStreamSignature> validatedSignatures = signatureHashMap.get(key);
 
-            if (canReachConsensus(validatedSignatures.size(), nodeIDPubKeyMap.size())) {
+            if (canReachConsensus(validatedSignatures.size(), nodeCount)) {
                 consensusCount += validatedSignatures.size();
                 validatedSignatures.forEach(s -> s.setStatus(SignatureStatus.CONSENSUS_REACHED));
             }

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/NodeSignatureVerifier.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/NodeSignatureVerifier.java
@@ -23,7 +23,6 @@ package com.hedera.mirror.importer.downloader;
 import com.google.common.collect.HashMultimap;
 import com.google.common.collect.Multimap;
 import com.google.common.collect.Sets;
-import com.google.common.collect.TreeMultimap;
 import java.security.PublicKey;
 import java.security.Signature;
 import java.util.Collection;
@@ -67,43 +66,18 @@ public class NodeSignatureVerifier {
      * @throws SignatureVerificationException
      */
     public void verify(Collection<FileStreamSignature> signatures) throws SignatureVerificationException {
-        Multimap<String, FileStreamSignature> signaturesByName = TreeMultimap.create();
-        signatures.forEach(s -> signaturesByName.put(s.getFile().getName(), s));
-        Collection<String> filenames = signaturesByName.keySet();
-
-        if (filenames.size() > 1) {
-            log.warn("Found {} unique filenames for stream interval: {}", filenames.size(), signatures);
-        }
-
-        for (String filename : filenames) {
-            if (verifyFileGroup(signaturesByName.get(filename))) {
-                return;
-            }
-        }
-
-        throw new SignatureVerificationException("Signature verification failed for files " + filenames + ": " + statusMap(signatures));
-    }
-
-    /**
-     * Since balance files can occasionally generate a file with a different timestamp from different nodes or a rogue
-     * node can send a bad filename, we group files into time buckets then within that bucket check if a particular
-     * filename reaches consensus.
-     *
-     * @param signatures grouped by filename
-     * @return whether this file was verified
-     * @throws SignatureVerificationException
-     */
-    private boolean verifyFileGroup(Collection<FileStreamSignature> signatures) {
         Multimap<String, FileStreamSignature> signatureHashMap = HashMultimap.create();
-        String filename = null;
+        String filename = !signatures.isEmpty() ? signatures.stream().findFirst().get().getFile().getName() : null;
         int consensusCount = 0;
-        boolean verified = false;
+
+        final long sigFileCount = signatures.size();
+        final long nodeCount = nodeIDPubKeyMap.size();
+        if (!consensusReached(sigFileCount, nodeCount)) {
+            throw new SignatureVerificationException("Require at least 1/3 signature files to reach consensus, got " +
+                    sigFileCount + " out of " + nodeCount + " for file " + filename + ": " + statusMap(signatures));
+        }
 
         for (FileStreamSignature fileStreamSignature : signatures) {
-            if (filename == null) {
-                filename = fileStreamSignature.getFile().getName();
-            }
-
             Pair<byte[], byte[]> hashAndSig = Utility.extractHashAndSigFromFile(fileStreamSignature.getFile());
             if (hashAndSig == null) {
                 continue;
@@ -124,20 +98,20 @@ public class NodeSignatureVerifier {
 
             if (consensusReached(validatedSignatures.size(), nodeIDPubKeyMap.size())) {
                 consensusCount += validatedSignatures.size();
-                validatedSignatures.stream().forEach(s -> s.setStatus(SignatureStatus.CONSENSUS_REACHED));
+                validatedSignatures.forEach(s -> s.setStatus(SignatureStatus.CONSENSUS_REACHED));
             }
         }
 
-        if (consensusCount == nodeIDPubKeyMap.size()) {
+        if (consensusCount == nodeCount) {
             log.debug("Verified signature file {} reached consensus", filename);
-            verified = true;
+            return;
         } else if (consensusCount > 0) {
             log.warn("Verified signature file {} reached consensus but with some errors: {}", filename,
                     statusMap(signatures));
-            verified = true;
+            return;
         }
 
-        return verified;
+        throw new SignatureVerificationException("Signature verification failed for file " + filename + ": " + statusMap(signatures));
     }
 
     /**

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/balance/BalanceDownloaderProperties.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/balance/BalanceDownloaderProperties.java
@@ -46,9 +46,6 @@ public class BalanceDownloaderProperties implements DownloaderProperties {
     @Min(1)
     private int batchSize = 15;
 
-    @NotNull
-    private Duration closeInterval = Duration.ofMinutes(15L);
-
     private boolean enabled = true;
 
     @NotNull

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/event/EventDownloaderProperties.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/event/EventDownloaderProperties.java
@@ -46,9 +46,6 @@ public class EventDownloaderProperties implements DownloaderProperties {
     @Min(1)
     private int batchSize = 100;
 
-    @NotNull
-    private Duration closeInterval = Duration.ofSeconds(5L);
-
     private boolean enabled = false;
 
     @NotNull

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/record/RecordDownloaderProperties.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/record/RecordDownloaderProperties.java
@@ -46,9 +46,6 @@ public class RecordDownloaderProperties implements DownloaderProperties {
     @Min(1)
     private int batchSize = 40;
 
-    @NotNull
-    private Duration closeInterval = Duration.ofSeconds(2L);
-
     private boolean enabled = true;
 
     @NotNull

--- a/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/downloader/AbstractDownloaderTest.java
+++ b/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/downloader/AbstractDownloaderTest.java
@@ -41,7 +41,6 @@ import java.time.Duration;
 import java.time.Instant;
 import java.util.Arrays;
 import java.util.List;
-import javax.sql.DataSource;
 import org.apache.commons.io.FileUtils;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
@@ -79,8 +78,6 @@ public abstract class AbstractDownloaderTest {
     protected Downloader downloader;
     protected Path validPath;
     protected MeterRegistry meterRegistry = new LoggingMeterRegistry();
-    @Mock(answer = Answers.RETURNS_SMART_NULLS)
-    protected DataSource dataSource;
     protected String file1;
     protected String file2;
 
@@ -123,6 +120,8 @@ public abstract class AbstractDownloaderTest {
     protected abstract Downloader getDownloader();
 
     protected abstract Path getTestDataDir();
+
+    protected abstract Duration getCloseInterval();
 
     boolean isSigFile(Path path) {
         return path.toString().endsWith("_sig");
@@ -322,25 +321,25 @@ public abstract class AbstractDownloaderTest {
     @Test
     @DisplayName("Different filenames, same interval, lower bound")
     void differentFilenamesSameIntervalLower() throws Exception {
-        differentFilenames(downloaderProperties.getCloseInterval().dividedBy(2L).negated());
+        differentFilenames(getCloseInterval().dividedBy(2L).negated());
     }
 
     @Test
     @DisplayName("Different filenames, same interval, upper bound")
     void differentFilenamesSameIntervalUpper() throws Exception {
-        differentFilenames(downloaderProperties.getCloseInterval().dividedBy(2L).minusNanos(1));
+        differentFilenames(getCloseInterval().dividedBy(2L).minusNanos(1));
     }
 
     @Test
     @DisplayName("Different filenames, previous interval")
     void differentFilenamesPreviousInterval() throws Exception {
-        differentFilenames(downloaderProperties.getCloseInterval().dividedBy(2L).negated().minusNanos(2));
+        differentFilenames(getCloseInterval().dividedBy(2L).negated().minusNanos(2));
     }
 
     @Test
     @DisplayName("Different filenames, next interval")
     void differentFilenamesNextInterval() throws Exception {
-        differentFilenames(downloaderProperties.getCloseInterval().dividedBy(2L));
+        differentFilenames(getCloseInterval().dividedBy(2L));
     }
 
     private void differentFilenames(Duration offset) throws Exception {
@@ -350,7 +349,7 @@ public abstract class AbstractDownloaderTest {
         Path basePath = fileCopier.getTo().resolve(type.getNodePrefix() + "0.0.3");
 
         // Construct a new filename with the offset added to the last valid file
-        long nanoOffset = downloaderProperties.getCloseInterval().plus(offset).toNanos();
+        long nanoOffset = getCloseInterval().plus(offset).toNanos();
         long timestamp = Utility.getTimestampFromFilename(file1) + nanoOffset;
         String baseFilename = Instant.ofEpochSecond(0, timestamp).toString().replace(':', '_') + type.getSuffix() + ".";
 

--- a/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/downloader/AbstractDownloaderTest.java
+++ b/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/downloader/AbstractDownloaderTest.java
@@ -342,6 +342,30 @@ public abstract class AbstractDownloaderTest {
         differentFilenames(getCloseInterval().dividedBy(2L));
     }
 
+    @Test
+    @DisplayName("Download and verify two group of files in the same bucket")
+    void downloadValidFilesInSameBucket() throws Exception {
+        // last valid downloaded file's timestamp is set to file1's timestamp - (I/2 + 1ns), so both file1 and file2
+        // will be in the bucket [lastTimestamp + I/2, lastTimestamp + 3*I/2). Note the interval I is set to twice of
+        // the difference between file1 and file2.
+        Duration interval = getCloseInterval().multipliedBy(2);
+        long file1Timestamp = Utility.getTimestampFromFilename(file1);
+        Instant file1Instant = Instant.ofEpochSecond(file1Timestamp / 1_000_000_000L,
+                file1Timestamp % 1_000_000_000L);
+        Instant lastFileInstant = file1Instant.minus(interval.dividedBy(2).plusNanos(1));
+        StreamType streamType = StreamType.fromFilename(file1);
+        String basename = lastFileInstant.toString().replace(':', '_');
+        String lastFileName = basename + streamType.getSuffix() + "." + streamType.getExtension();
+
+        doReturn(lastFileName).when(applicationStatusRepository).findByStatusCode(downloader.getLastValidDownloadedFileKey());
+
+        fileCopier.copy();
+        downloader.download();
+
+        verifyForSuccess();
+        assertThat(downloaderProperties.getSignaturesPath()).doesNotExist();
+    }
+
     private void differentFilenames(Duration offset) throws Exception {
         // Copy all files and modify only node 0.0.3's files to have a different timestamp
         StreamType type = downloaderProperties.getStreamType();

--- a/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/downloader/balance/AccountBalancesDownloaderTest.java
+++ b/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/downloader/balance/AccountBalancesDownloaderTest.java
@@ -21,6 +21,7 @@ package com.hedera.mirror.importer.downloader.balance;
  */
 
 import java.nio.file.Path;
+import java.time.Duration;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
@@ -48,6 +49,11 @@ public class AccountBalancesDownloaderTest extends AbstractDownloaderTest {
     @Override
     protected Path getTestDataDir() {
         return Path.of("accountBalances");
+    }
+
+    @Override
+    protected Duration getCloseInterval() {
+        return Duration.ofMinutes(15L);
     }
 
     @BeforeEach

--- a/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/downloader/event/EventFileDownloaderTest.java
+++ b/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/downloader/event/EventFileDownloaderTest.java
@@ -22,6 +22,7 @@ package com.hedera.mirror.importer.downloader.event;
 
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.time.Duration;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
@@ -52,6 +53,11 @@ public class EventFileDownloaderTest extends AbstractLinkedStreamDownloaderTest 
     @Override
     protected Path getTestDataDir() {
         return Paths.get("eventsStreams", "v3");
+    }
+
+    @Override
+    protected Duration getCloseInterval() {
+        return Duration.ofSeconds(5L);
     }
 
     @Test

--- a/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/downloader/record/RecordFileDownloaderTest.java
+++ b/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/downloader/record/RecordFileDownloaderTest.java
@@ -63,11 +63,15 @@ public class RecordFileDownloaderTest extends AbstractLinkedStreamDownloaderTest
         return Paths.get("recordstreams", "v2");
     }
 
+    @Override
+    protected Duration getCloseInterval() {
+        return Duration.ofSeconds(5L);
+    }
+
     @BeforeEach
     void beforeEach() {
         file1 = "2019-08-30T18_10_00.419072Z.rcd";
         file2 = "2019-08-30T18_10_05.249678Z.rcd";
-        downloaderProperties.setCloseInterval(Duration.ofSeconds(5)); // Test rcd files use older 5s interval
     }
 
     @Test

